perfkitbenchmarker/linux_benchmarks/cloud_spanner_ycsb_benchmark.py

```diff
@@ -0,0 +1,185 @@
+    'https://www.googleapis.com/auth/spanner.data')
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string('table',
+                    'usertable',
+                    'The table name used in the YCSB benchmark.')
+flags.DEFINE_integer('zeropadding',
+                     12,
+                     'The zero padding used in the YCSB benchmark.')
+flags.DEFINE_string('cloud_spanner_host',
+                    None,
+                    'The Cloud Spanner host used in the YCSB benchmark.')
+flags.DEFINE_string('cloud_spanner_instance',
+                    'ycsb-' + getpass.getuser(),
+                    'The Cloud Spanner instance used in the YCSB benchmark.')
+flags.DEFINE_string('cloud_spanner_database',
+                    'ycsb',
+                    'The Cloud Spanner database used in the YCSB benchmark.')
+flags.DEFINE_string('cloud_spanner_readmode',
+                    'query',
+                    'The Cloud Spanner read mode used in the YCSB benchmark.')
+flags.DEFINE_integer('cloud_spanner_boundedstaleness',
+                     0,
+                     'The Cloud Spanner bounded staleness used in the YCSB '
+                     'benchmark.')
+flags.DEFINE_integer('cloud_spanner_batchinserts',
+                     1,
+                     'The Cloud Spanner batch inserts used in the YCSB '
+                     'benchmark.')
+flags.DEFINE_string('cloud_spanner_custom_ycsb_release',
+                    None,
+                    'If provided, the URL of a custom YCSB release')
+flags.DEFINE_string('cloud_spanner_desp',
+                    'YCSB',
+                    'The description of the Cloud Spanner instance.')
+flags.DEFINE_integer('cloud_spanner_nodes',
+                     1,
+                     'The number of nodes for the Cloud Spanner instance.')
+flags.DEFINE_string('cloud_spanner_config',
+                    'regional-us-central1',
+                    'The config for the Cloud Spanner instance.')
+flags.DEFINE_string('cloud_spanner_ddl',
+                    """CREATE TABLE usertable (
+                    id STRING(MAX), field0 STRING(MAX), field1 STRING(MAX),
+                    field2 STRING(MAX), field3 STRING(MAX), field4 STRING(MAX),
+                    field5 STRING(MAX), field6 STRING(MAX), field7 STRING(MAX),
+                    field8 STRING(MAX), field9 STRING(MAX),
+                    ) PRIMARY KEY(id)""",
+                    'The schema DDL for the Cloud Spanner database.')
+
+
+def GetConfig(user_config):
+  config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+  if FLAGS['ycsb_client_vms'].present:
+    config['vm_groups']['default']['vm_count'] = FLAGS.ycsb_client_vms
+  return config
+
+
+def CheckPrerequisites(benchmark_config):
+  for scope in REQUIRED_SCOPES:
+    if scope not in FLAGS.gcloud_scopes:
+      raise ValueError('Scope {0} required.'.format(scope))
+
+
+def Prepare(benchmark_spec):
+  benchmark_spec.always_call_cleanup = True
```

Comment:
This is needed to delete the cloud spanner instance after the test is done. Otherwise I'll need to provide a way to bypass instance creation in the Prepare() method.